src/raftstore/store/peer.rs

```diff
@@ -414,6 +417,24 @@ impl Peer {
         Ok(peer)
     }
 
+    pub fn register_apply_delegate(&self) {
+        self.apply_scheduler
+            .schedule(ApplyTask::register(self))
+            .unwrap();
+    }
+
+    pub fn register_read_delegate(&self) {
+        self.read_scheduler
+            .schedule(ReadTask::register(self))
+            .unwrap();
+    }
+
+    pub fn deregister_read_delegate(&self) {
+        self.read_scheduler
```

Comment:
do we need to expire the remote lease forcibly here?

Reply:
It's not necessary, because its remote lease must be expired before deregister.